### DISCUSSION TITLE
Custom Default Library

### DISCUSSION THREAD

**Discussion by lmajhi** [jgraph/drawio#2574] [Q&A]
I’m using the fork of the repo to run a local setup. 
I have the xml file of custom shape library. How should i go about making changes to the code base, to include the custom shape library on the Sidebar?

**Comment by Peeeet** [comment]
Unfortunately there is no officially supported way to do this and no "developer documentation" if you wanted to dig into the code, but if anyone did figure out a way to do it I (and probably others) would gladly sponsor them on GitHub or BountySource or something.

**Reply by lmajhi** [reply]
I figured that if we can load custom libraries, if we add `{"defaultCustomLibraries": ["A1"]}` for `window.DRAWIO_CONFIG` in `js/PreConfig.js` file. The issue here being , how to get ids, for eg `A1` for the library from the xml. Any idea, would help.

**Comment by lmajhi** [comment]
@Peeeet : After much hit and trial, and reading the documentation , i seem to have figured it.  Please have a look https://medium.com/@lakshyaing/adding-external-library-in-draw-io-552b6bcd9e49

**Reply by Peeeet** [reply]
Interesting yeah, I assumed there's maybe some way to do it using the "plugin" functionality of Diagrams.net ( something to do with the "p1" plugin example at https://app.diagrams.net/plugins/p1.js ) but then you would have to add the `&p=encodedPluginUrl` parameter or something onto every url you used.

I had cloned drawio a while back and then immediately got lost in the woods, so I might try diving back into this also now that I'm digging into a bunch of self-hosting stuff these days too.

Thanks for posting this, I'll hit you up if I'm actually able to get it working or make more progress too! (PS the phrases are "hit or miss" as well as "trial and error" for future reference :-)
